### ISSUE TITLE
gcc48 fails to build on 10.11

### ISSUE THREAD

**Issue by alinpopa** [Homebrew/homebrew-versions#984]
```
~ % brew install gcc48
==> Installing gcc48 from homebrew/homebrew-versions
==> Downloading http://ftpmirror.gnu.org/gcc/gcc-4.8.4/gcc-4.8.4.tar.bz2
Already downloaded: /Library/Caches/Homebrew/gcc48-4.8.4.tar.bz2
==> Downloading https://gcc.gnu.org/bugzilla/attachment.cgi?id=35773
Already downloaded: /Library/Caches/Homebrew/gcc48--patch-db4966ade190fff4ed39976be8d13e84839098711713eff1d08920d37a58f5ec.cgi
==> Patching
patching file gcc/config/darwin.c
Hunk #1 succeeded at 1204 (offset -55 lines).
==> ../configure --build=x86_64-apple-darwin15.0.0 --prefix=/usr/local/Cellar/gcc48/4.8.4 --libdir=/usr/local/Cellar/gcc48/4.8.4/lib/gcc/4.8 --enable-languages=c,c++,objc,obj-c++ --program-suffix=-4.8 --with-gmp=/usr/local/opt/gmp4 --with-mpfr=/usr/local/opt/mpfr2 --with-mpc=/u
==> make bootstrap
Last 15 lines from /Users/alin/Library/Logs/Homebrew/gcc48/02.make:
   #if __has_attribute(availability)
                      ^
make[4]: *** [sanitizer_allocator.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
mv -f .deps/sanitizer_linux.Tpo .deps/sanitizer_linux.Plo
mv -f .deps/sanitizer_platform_limits_posix.Tpo .deps/sanitizer_platform_limits_posix.Plo
mv -f .deps/sanitizer_flags.Tpo .deps/sanitizer_flags.Plo
mv -f .deps/sanitizer_common.Tpo .deps/sanitizer_common.Plo
mv -f .deps/sanitizer_posix.Tpo .deps/sanitizer_posix.Plo
mv -f .deps/sanitizer_libc.Tpo .deps/sanitizer_libc.Plo
mv -f .deps/sanitizer_mac.Tpo .deps/sanitizer_mac.Plo
make[3]: *** [all-recursive] Error 1
make[2]: *** [all-stage1-target-libsanitizer] Error 2
make[1]: *** [stage1-bubble] Error 2
make: *** [bootstrap] Error 2
couldn't understand kern.osversion `15.0.0'

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/homebrew):
  https://github.com/homebrew/homebrew-versions/issues

Warning:
~ %
```
```
~ % brew config
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Homebrew/homebrew.git
HEAD: 41132af5f4c9647e97f84ebaba824320ed09389c
Last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: 8-core 64-bit haswell
OS X: 10.11-x86_64
Xcode: 7.0.1
CLT: 7.0.0.0.1.1441394355
couldn't understand kern.osversion `15.0.0'
GCC-4.2: build 5666
Clang: 7.0 build 700
X11: 2.7.7 => /opt/X11
System Ruby: 2.0.0-p645
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /Users/alin/.rvm/rubies/ruby-2.0.0-p643/bin/ruby
Java: 1.8.0_60, 1.8.0_40, 1.8.0_31, 1.7.0_75
~ %
```
```
~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: The filesystem on / appears to be case-sensitive.
The default OS X filesystem is case-insensitive. Please report any apparent problems.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
    /usr/local/lib/libtcl8.6.dylib
    /usr/local/lib/libtk8.6.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
    /usr/local/include/fakemysql.h
    /usr/local/include/fakepq.h
    /usr/local/include/fakesql.h
    /usr/local/include/itcl.h
    /usr/local/include/itcl2TclOO.h
    /usr/local/include/itclDecls.h
    /usr/local/include/itclInt.h
    /usr/local/include/itclIntDecls.h
    /usr/local/include/itclMigrate2TclCore.h
    /usr/local/include/itclTclIntStubsFcn.h
    /usr/local/include/mysqlStubs.h
    /usr/local/include/odbcStubs.h
    /usr/local/include/pqStubs.h
    /usr/local/include/tcl.h
    /usr/local/include/tclDecls.h
    /usr/local/include/tclOO.h
    /usr/local/include/tclOODecls.h
    /usr/local/include/tclPlatDecls.h
    /usr/local/include/tclThread.h
    /usr/local/include/tclTomMath.h
    /usr/local/include/tclTomMathDecls.h
    /usr/local/include/tdbc.h
    /usr/local/include/tdbcDecls.h
    /usr/local/include/tdbcInt.h
    /usr/local/include/tk.h
    /usr/local/include/tkDecls.h
    /usr/local/include/tkPlatDecls.h

Warning: Unbrewed .pc files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected .pc files:
    /usr/local/lib/pkgconfig/tcl.pc
    /usr/local/lib/pkgconfig/tk.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
    /usr/local/lib/libtclstub8.6.a
    /usr/local/lib/libtkstub8.6.a

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
    gcc5
~ %
```

I've noticed the same issue happening for gcc46 as well.
Also, exactly the same thing happened before upgrading to El Capitan, on Yosemite, after installing xcode 7.

**Comment by ohthehugemanatee** [comment]
I had the same problem installing gcc47. I got advice from another thread that this might be the build getting confused, and using Apple's native gcc42. I got past the error by running ``` brew uninstall apple-gcc42 ```

But now the compile segfaults, like this:

``` bash
~$ brew install gcc47
brew install gcc47
==> Installing gcc47 from homebrew/homebrew-versions
==> Downloading http://ftpmirror.gnu.org/gcc/gcc-4.7.4/gcc-4.7.4.tar.bz2
Already downloaded: /Library/Caches/Homebrew/gcc47-4.7.4.tar.bz2
==> Downloading https://gcc.gnu.org/bugzilla/attachment.cgi?id=35773
Already downloaded: /Library/Caches/Homebrew/gcc47--patch-db4966ade190fff4ed39976be8d13e84839098711713eff1d08920d37a58f5ec.cgi
==> Downloading https://trac.macports.org/export/126996/trunk/dports/lang/gcc48/files/patch-10.10.diff
Already downloaded: /Library/Caches/Homebrew/gcc47--patch-4fb0ededa7b8105c3bdffa15469b589b272b7788.diff
==> Patching
patching file gcc/config/darwin.c
Hunk #1 succeeded at 1197 (offset -62 lines).
patching file gcc/config/darwin-c.c
patching file gcc/config/darwin-driver.c
==> ../configure --build=x86_64-apple-darwin15.0.0 --prefix=/usr/local/Cellar/gcc47/4.7.4 --enable-languages=c,c++,objc,obj-c++ --program-suffix=-4.7 --with-gmp=/usr/local/opt/gmp4 --with-mpfr=/usr/local/
==> make bootstrap
Last 15 lines from /Users/ohthehugemanatee/Library/Logs/Homebrew/gcc47/02.make:
	   libcommon.a ../libcpp/libcpp.a  -liconv ../libiberty/libiberty.a ../libdecnumber/libdecnumber.a 
clang   -g -fkeep-inline-functions -DIN_GCC   -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wstrict-prototypes -Wmissing-prototypes -Wmissing-format-attribute -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings -Wold-style-definition -Wc++-compat -fno-common  -DHAVE_CONFIG_H  -o cpp gcc.o ggc-none.o \
	  cppspec.o driver-i386.o darwin-driver.o libcommon-target.a \
	   libcommon.a ../libcpp/libcpp.a  -liconv ../libiberty/libiberty.a ../libdecnumber/libdecnumber.a 
clang   -g -fkeep-inline-functions -DIN_GCC   -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wstrict-prototypes -Wmissing-prototypes -Wmissing-format-attribute -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings -Wold-style-definition -Wc++-compat -fno-common  -DHAVE_CONFIG_H  -o g++ \
	  gcc.o ggc-none.o g++spec.o driver-i386.o darwin-driver.o libcommon-target.a \
	   libcommon.a ../libcpp/libcpp.a  -liconv ../libiberty/libiberty.a ../libdecnumber/libdecnumber.a 
/private/tmp/gcc4720151008-68920-ui0kyk/gcc-4.7.4/build/./gcc/xgcc -B/private/tmp/gcc4720151008-68920-ui0kyk/gcc-4.7.4/build/./gcc/ -dumpspecs > tmp-specs
/bin/sh: line 1: 91501 Segmentation fault: 11  /private/tmp/gcc4720151008-68920-ui0kyk/gcc-4.7.4/build/./gcc/xgcc -B/private/tmp/gcc4720151008-68920-ui0kyk/gcc-4.7.4/build/./gcc/ -dumpspecs > tmp-specs
make[3]: *** [specs] Error 139
make[3]: *** Waiting for unfinished jobs....
rm gcc.pod
make[2]: *** [all-stage1-gcc] Error 2
make[1]: *** [stage1-bubble] Error 2
make: *** [bootstrap] Error 2

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/homebrew):
  https://github.com/homebrew/homebrew-versions/issues

Warning: 
~$
```
(yes, the warning is empty)

Do you get the same? For reference

``` bash
~$ brew config
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Homebrew/homebrew
HEAD: b6fe8bcc8695a0f0158758355310f6052741689c
Last commit: 24 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit haswell
OS X: 10.11-x86_64
Xcode: 7.0.1
CLT: 7.0.0.0.1.1441394355
Clang: 7.0 build 700
X11: N/A
System Ruby: 2.0.0-p645
Perl: /Applications/MAMP/Library/bin/perl
Python: /Applications/MAMP/Library/bin/python => /Applications/MAMP/Library/bin/python2.7
Ruby: /Users/ohthehugemanatee/.rvm/rubies/ruby-2.1.2/bin/ruby
Java: N/A
```

``` bash
~$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
    /Applications/MAMP/Library/bin/curl-config
    /Applications/MAMP/Library/bin/freetype-config
    /Applications/MAMP/Library/bin/icu-config
    /Applications/MAMP/Library/bin/libmcrypt-config
    /Applications/MAMP/Library/bin/libpng-config
    /Applications/MAMP/Library/bin/libpng16-config
    /Applications/MAMP/Library/bin/Magick++-config
    /Applications/MAMP/Library/bin/Magick-config
    /Applications/MAMP/Library/bin/MagickCore-config
    /Applications/MAMP/Library/bin/MagickWand-config
    /Applications/MAMP/Library/bin/pcre-config
    /Applications/MAMP/Library/bin/python-config
    /Applications/MAMP/Library/bin/python2-config
    /Applications/MAMP/Library/bin/python2.7-config
    /Applications/MAMP/Library/bin/sablot-config
    /Applications/MAMP/Library/bin/Wand-config
    /Applications/MAMP/Library/bin/xml2-config
    /Applications/MAMP/Library/bin/xslt-config
    /Applications/MAMP/Library/bin/yaz-config

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
    /usr/local/include/node/android-ifaddrs.h
    /usr/local/include/node/ares.h
    /usr/local/include/node/ares_version.h
    /usr/local/include/node/libplatform/libplatform.h
    /usr/local/include/node/nameser.h
    /usr/local/include/node/node.h
    /usr/local/include/node/node_buffer.h
    /usr/local/include/node/node_internals.h
    /usr/local/include/node/node_object_wrap.h
    /usr/local/include/node/node_version.h
    /usr/local/include/node/openssl/opensslconf.h
    /usr/local/include/node/pthread-fixes.h
    /usr/local/include/node/smalloc.h
    /usr/local/include/node/stdint-msvc2008.h
    /usr/local/include/node/tree.h
    /usr/local/include/node/uv-aix.h
    /usr/local/include/node/uv-bsd.h
    /usr/local/include/node/uv-darwin.h
    /usr/local/include/node/uv-errno.h
    /usr/local/include/node/uv-linux.h
    /usr/local/include/node/uv-sunos.h
    /usr/local/include/node/uv-threadpool.h
    /usr/local/include/node/uv-unix.h
    /usr/local/include/node/uv-version.h
    /usr/local/include/node/uv-win.h
    /usr/local/include/node/uv.h
    /usr/local/include/node/v8-debug.h
    /usr/local/include/node/v8-platform.h
    /usr/local/include/node/v8-profiler.h
    /usr/local/include/node/v8-testing.h
    /usr/local/include/node/v8-util.h
    /usr/local/include/node/v8.h
    /usr/local/include/node/v8config.h
    /usr/local/include/node/v8stdint.h
    /usr/local/include/node/zconf.h
    /usr/local/include/node/zlib.h
```

**Comment by alinpopa** [comment]
Yes, I'm getting the same issue as you; also, if removing `apple-gcc42` makes no difference.

**Comment by joshRpowell** [comment]
I get the same error.
```
➜  ~  brew install gcc48
==> Installing gcc48 from homebrew/homebrew-versions
==> Downloading http://ftpmirror.gnu.org/gcc/gcc-4.8.4/gcc-4.8.4.tar.bz2
Already downloaded: /Library/Caches/Homebrew/gcc48-4.8.4.tar.bz2
==> Downloading https://gcc.gnu.org/bugzilla/attachment.cgi?id=35773
Already downloaded: /Library/Caches/Homebrew/gcc48--patch-db4966ade190fff4ed39976be8d13e84839098711713eff1d08920d37a58f5ec.cgi
==> Patching
patching file gcc/config/darwin.c
Hunk #1 succeeded at 1204 (offset -55 lines).
==> ../configure --build=x86_64-apple-darwin15.0.0 --prefix=/usr/local/Cellar/gcc48/4.8.4 --libdir=/usr/local/Cellar/gcc48/4.8
==> make bootstrap
Last 15 lines from /Users/jpowell/Library/Logs/Homebrew/gcc48/02.make:
/usr/include/Availability.h:219:22: error: missing binary operator before token "("
   #if __has_attribute(availability)
                      ^
/usr/include/Availability.h:241:22: error: missing binary operator before token "("
   #if __has_attribute(availability)
                      ^
make[4]: *** [sanitizer_allocator.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
mv -f .deps/sanitizer_flags.Tpo .deps/sanitizer_flags.Plo
mv -f .deps/sanitizer_common.Tpo .deps/sanitizer_common.Plo
mv -f .deps/sanitizer_libc.Tpo .deps/sanitizer_libc.Plo
make[3]: *** [all-recursive] Error 1
make[2]: *** [all-stage1-target-libsanitizer] Error 2
make[1]: *** [stage1-bubble] Error 2
make: *** [bootstrap] Error 2

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/homebrew):
  https://github.com/homebrew/homebrew-versions/issues

Warning:

**Comment by doapp-ryanp** [comment]
+1

**Comment by dunn** [comment]
Can one of you try compiling 4.8.4 outside of Homebrew and see if it has the same problem?

**Comment by alinpopa** [comment]
Tried that as well, but no real success:

**Comment by prasant94** [comment]
same issue

**Comment by doapp-ryanp** [comment]
I just installed gcc49 via homebrew and things are working well now.

**Comment by alinpopa** [comment]
gcc49 indeed, works; gcc48 still has the same issue.

**Comment by dunn** [comment]
Sounds like 4.8.4 isn't compatible with El Capitan; if someone's willing to open a ticket with the GCC developers, that'd be great!

**Comment by DomT4** [comment]
Merged #1000 to fix this.
